cmd/geth/main.go

```diff
@@ -457,24 +450,23 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 		if ctx.GlobalString(utils.SyncModeFlag.Name) == "light" {
```

Comment:
I think everything in this block is code that only works for full nodes.
We should create an auxiliary function for setting up `*eth.Ethereum` specifically
and then put a type assertion for `*eth.Ethereum` here.